aspnet/fundamentals/caching.rst

```diff
@@ -1,8 +1,157 @@
-.. include:: /../common/stub-topic.txt
+In Memory Caching
+=================
 
-|stub-icon| Caching
-===================
+By `Steve Smith`_
 
-.. include:: /../common/stub-notice.txt
+Caching involves keeping a copy of data in a location that can be accessed more quickly than the source 
+data. ASP.NET 5 has rich support for caching in a variety of ways, including keeping data in memory on the 
+local server, which is referred to as *in memory caching*.
 
-.. _issue: https://github.com/aspnet/Docs/issues/69
\ No newline at end of file
+In this article:
+	- `Caching Basics`_
+	- `Configuring In Memory Caching`_
+	- `Reading and Writing to a Memory Cache`_
+	- `Cache Dependencies and Callbacks`_
+
+`View or download sample from GitHub <https://github.com/aspnet/Docs/tree/1.0.0-beta8/aspnet/fundamentals/caching/sample>`_.
+
+Caching Basics
+--------------
+Caching can dramatically improve the performance and scalability of ASP.NET applications, by eliminating unnecessary requests to external data sources for data that hasn't changed since it was last requested. ASP.NET supports several different kinds of caches, the simplest of which is represented by the `IMemoryCache <https://github.com/aspnet/Caching/blob/dev/src/Microsoft.Extensions.Caching.Abstractions/IMemoryCache.cs>`_ interface, which represents a cache stored in the memory of the local web server.
+
+An in-memory cache is stored in the memory of a single server hosting an ASP.NET application. Thus, if an application is hosted by multiple servers in a web farm or cloud hosting environment, it is possible that different servers will have different values in their local in-memory caches. If this is an issue for the application, a :doc:`distributed-cache` may be a better alternative.
+
+.. note:: Caching in all forms (in-memory or distributed, including session state) involves making a copy of data in order to optimize performance. The copied data should be considered ephemeral - it could disappear at any time. You should always write your application such that it can use cached data if it's available, but otherwise will work correctly using the underlying data source.
+
+Configuring In Memory Caching
+-----------------------------
+To get started working with an in memory cache in your ASP.NET application, you should first add the following dependencies to your ``project.json`` file:
+
+.. literalinclude:: caching/sample/src/CachingSample/project.json
+	:linenos:
+	:lines: 5-10
+	:emphasize-lines: 4-5
+
+Caching in ASP.NET 5 is a *service* that should be referenced from your application via :doc:`dependency-injection`. To register the caching service and make it available within your application, add the following line to your ``ConfigureServices`` method in ``Startup``:
```

Comment:
However, that will have negative coupling and testability effects (in particular if you want to test non-trivial behavior that occurs when the cache is or isn't hit), so the recommendation should be to treat it as a service, IMO.